tests/core/engine_adapter/test_duckdb.py

```diff
@@ -69,3 +72,59 @@ def test_transaction(adapter: EngineAdapter, duck_conn):
     except Exception:
         pass
     assert duck_conn.execute("SELECT * FROM test_table").fetchall() == [(1,)]
+
+
+def test_merge(mocker: MockerFixture):
+    connection_mock = mocker.NonCallableMock()
+    cursor_mock = mocker.Mock()
+    connection_mock.cursor.return_value = cursor_mock
+    temp_table_mock = mocker.patch("sqlmesh.core.engine_adapter.base.EngineAdapter._get_temp_table")
+    temp_table_mock.return_value = exp.to_table("temporary")
+
+    adapter = DuckDBEngineAdapter(lambda: connection_mock)  # type: ignore
+    adapter.merge(
+        target_table="target",
+        source_table=parse_one("SELECT id, ts, val FROM source"),
+        columns_to_types={
+            "id": exp.DataType(this=exp.DataType.Type.INT),
+            "ts": exp.DataType(this=exp.DataType.Type.TIMESTAMP),
+            "val": exp.DataType(this=exp.DataType.Type.INT),
+        },
+        unique_key=["id"],
+    )
+
+    cursor_mock.execute.assert_has_calls(
+        [
+            call("CREATE TABLE temporary AS SELECT id, ts, val FROM source"),
+            call("DELETE FROM target WHERE id IN (SELECT id FROM temporary)"),
+            call(
+                "INSERT INTO target (id, ts, val) (SELECT DISTINCT ON (id) id, ts, val FROM temporary)"
```

Comment:
Is distinct on widely supported? I wasn't aware of this syntax.

Reply:
No, it isn't (as I found out last night). It came from postgres and a few other dbs have picked it up.

sqlglot will translate from dictinct on to syntax that works for dbs that don't support.